api/scpca_portal/metadata_file.py

```diff
@@ -92,10 +92,25 @@ def write_metadata_dicts(list_of_dicts: List[Dict], output_file_path: str, **kwa
     Writes a list of dictionaries to a csv-like file.
     Optional modifiers to the csv.DictWriter can be passed to function as kwargs.
     """
-    kwargs["fieldnames"] = kwargs.get("fieldnames", utils.get_keys_from_dicts(list_of_dicts))
+    kwargs["fieldnames"] = kwargs.get(
+        "fieldnames", utils.get_sorted_field_names(utils.get_keys_from_dicts(list_of_dicts))
+    )
     kwargs["delimiter"] = kwargs.get("delimiter", common.TAB)
 
+    sorted_list_of_dicts = (
+        sorted(
+            list_of_dicts,
+            key=lambda k: (
+                k[common.METADATA_ROW_SORT_ORDER[0]],
+                k[common.METADATA_ROW_SORT_ORDER[1]],
+                k[common.METADATA_ROW_SORT_ORDER[2]],
+            ),
+        )
+        if any(common.METADATA_ROW_SORT_ORDER[0] in d for d in list_of_dicts)
+        else list_of_dicts
+    )
```

Comment:
Could you test this, I think this makes sense as a fallback until we standardize which keys go into all of the metadata files.
```suggestion
        sorted(
            list_of_dicts,
            key=lambda k: (
                k.get(common.PROJECT_ID_KEY),
                k.get(common.SAMPLE_ID_KEY),
                k.get(common.LIBRARY_ID_KEY)
            ),
        )
    )
```